app/Http/Controllers/UserController.php

```diff
@@ -4,54 +4,42 @@
 
 use App\Models\User;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Str;
 
 class UserController extends Controller
 {
     public function index()
     {
-        // TASK: turn this SQL query into Eloquent
-        // select * from users
-        //   where email_verified_at is not null
-        //   order by created_at desc
-        //   limit 3
-
-        $users = User::all(); // replace this with Eloquent statement
+        $users = User::active()->orderByDesc('created_at')->take(3)->get();
 
         return view('users.index', compact('users'));
     }
 
     public function show($userId)
     {
-        $user = NULL; // TASK: find user by $userId or show "404 not found" page
+        $user = User::findOrFail($userId); // TASK: find user by $userId or show "404 not found" page
 
         return view('users.show', compact('user'));
     }
 
     public function check_create($name, $email)
     {
-        // TASK: find a user by $name and $email
-        //   if not found, create a user with $name, $email and random password
-        $user = NULL;
-
+        $user = User::firstOrCreate(compact('name', 'email'), ['password' => Hash::make(Str::random())]);
```

Comment:
You can use the new password function, check it: https://www.amitmerchant.com/laravel-10-password-generator/